website/docs/docs/build/incremental-microbatch.md

```diff
@@ -280,6 +282,126 @@ For now, dbt assumes that all values supplied are in UTC:
 
 While we may consider adding support for custom time zones in the future, we also believe that defining these values in UTC makes everyone's lives easier.
 
+## Parallel batch execution
+
+The microbatch strategy offers the benefit of updating a model in smaller, more manageable batches. 
+
+Parallel batch execution means that multiple batches are processed at the same time using the `concurrent_batches` config, instead of one after the other (sequentially) for faster processing of your microbatch models. 
+
+For example, if you have a microbatch model with 12 batches, you can execute those batches to run in parallel. Specifically they'll run in parallel limited by the number of [available threads](/docs/running-a-dbt-project/using-threads).
+
+### Prerequisites
+
+To enable parallel execution, you must meet the following conditions:
+
+- You use the following supported adapters:
+  - Snowflake
+  - Databricks
+  - More adapters coming soon!
+- The relation in the data warehouse for the model doesn't exist
+- You meet [additional conditions](#how-parallel-batch-execution-works) mentioned in the next section
+
+### How parallel batch execution works
+
+A batch can only run in parallel if:
+
+| Step | Condition     |  Parallel execution   | Sequential execution|
+| ---- | ---------------| :------------------: | :----------: |
+| 1.   | **Not** the first batch |  ✅         | -            |
+| 2.   | **Not** the last batch  |  ✅         | -            |
+| 3.   | [Adapter supports](#prerequisites) parallel batches | ✅  | -         |
+| 4.   | `concurrent_batches` set to `True`  |  ✅     |   -  |
+| 5.   | `concurrent_batches` set to `False` |     -   |   ✅ |
+
+
+After checking for 1, 2, and 3 in the previous table &mdash; and if `concurrent_batches` value isn't set, dbt will intelligently auto-detect if the model invokes the [`{{ this }}`](/reference/dbt-jinja-functions/this) Jinja function. If it references `{{ this }}`, the batches will run sequentially since  `{{ this }}` represents the database of the current model and referencing the same relation causes conflict. 
+
+Otherwise, if the `concurrent_batches` value isn't set _and_ `{{ this }}` isn't detected (and other conditions are met), the batches will run in parallel.  
```

Comment:
Hiya @QMalcolm 

I also wanted to ask about link 317. In the git issue, it advises: 

"If neither (1) nor (2) was hit, then we check if a config concurrent_batches is set for the model. If the value for that config is True then we run the batches in parallel, if False we run the batches sequentially.

If however concurrent_batches is None (i.e. not set), then we check if the model jinja contains a reference to this. If it references this then we run the batches sequentially. Otherwise, we run them in parallel."

Whereas, from your notes, it advises:

"After [1], [2], and [3] we check if the [this jinja function](https://docs.getdbt.com/reference/dbt-jinja-functions/this) is invoked in the model. If this is used, then the batch will be run sequentially, as it may be that your batch depends on the existence of prior batches. If this isn't used, the batch will be run in parallel.

You can override the check for this by setting concurrent_batches to either True or False. If set to False, the batch will be run sequentially. If set to True the batch will be run in parallel (assuming [1], [2], and [3])..."

I just wanted to double check if they meant the same thing as I was a little confused.

Kind Regards
Natalie

Reply:
Those two things are saying the same thing. The first one walks through the actual logical order of operations that happens in core under. The second is framed from how the end user should think about it. Basically, the user generally _shouldn't think about or set_ `concurrent_batches`. Core does its best to _automatically detect_ whether a batch can be run in parallel or not. The config `concurrent_batches` is an "escape hatch" to allow the user to say "I actually think these batches should/shouldn't be run in parallel". `concurrent_batches` isn't a _gate_ but instead an _override_.